includes/class-wc-google-gtag-js.php

```diff
@@ -201,26 +260,26 @@ function gtag(){dataLayer.push(arguments);}
 	 * @return string
 	 */
 	protected function add_transaction_enhanced( $order ) {
-		// Order items
-		$items = "[";
-		if ( $order->get_items() ) {
-			foreach ( $order->get_items() as $item ) {
-				$items .= self::add_item( $order, $item );
+		$event_items = array();
+		$order_items = $order->get_items();
+		if ( ! empty( $order_items ) ) {
+			foreach ( $order_items as $item ) {
+				$event_items += self::add_item( $order, $item );
 			}
 		}
-		$items .= "]";
-
-		$code = self::tracker_var() . "( 'event', 'purchase', {
-			'transaction_id': '" . esc_js( $order->get_order_number() ) . "',
-			'affiliation': '" . esc_js( get_bloginfo( 'name' ) ) . "',
-			'value': '" . esc_js( $order->get_total() ) . "',
-			'tax': '" . esc_js( $order->get_total_tax() ) . "',
-			'shipping': '" . esc_js( $order->get_total_shipping() ) . "',
-			'currency': '" . esc_js( $order->get_currency() ) . "',
-			'items': {$items},
-		} );";
-
-		return $code;
+
+		return self::get_event_code(
+			'purchase',
+			array(
+				'transaction_id' => $order->get_order_number(),
+				'affiliation'    => get_bloginfo( 'name' ),
+				'value'          => $order->get_total(),
+				'tax'            => $order->get_total_tax(),
+				'shipping'       => $order->get_total_shipping(),
+				'currency'       => $order->get_currency(),
+				'items'          => array( $event_items ),
```

Comment:
As mentioned above this should not be wrapped in an array.

```suggestion
				'items'          => $event_items,
```